modules/end-user-portal/src/Pages/Search/Detail.cshtml

```diff
@@ -0,0 +1,118 @@
+@page
+@model GpConnect.NationalDataSharingPortal.EndUserPortal.Pages.Search.DetailModel
+@using GpConnect.NationalDataSharingPortal.EndUserPortal.Helpers.Enumerations
+@using Microsoft.Extensions.Primitives
+@inject Microsoft.AspNetCore.Http.IHttpContextAccessor HttpContextAccessor
+@{
+    var source = DetailViewSource.Search;
```

Comment:
Can we not inject HttpContextAccessor into the base class (BaseModel) then populate BackPartialModel from properties in the DetailModel? It feels like this code should sit in the code behind rather than the Razor page

Reply:
I've refactored this code as well